src/layout/section/heroSection/index.unit.tsx

```diff
@@ -20,4 +20,18 @@ describe('HeroSection', () => {
     const button = wrapper.find('a')
     expect(button.text()).toContain('button text')
   })
+
+  it('should render a serch form', () => {
+    const props: HeroSectionProps = {
+      ...defaultProps,
+      // We don't need to use the _real_ `SearchForm` component as it is only used as a node.
+      searchForm: <form>Search Form</form>,
+    }
+
+    const wrapper = mount(<HeroSection {...props} />)
+    expect(wrapper.find('h1').text()).toContain('hero text')
```

Comment:
Avoid select items by tag (implementation details...)

Reply:
I agree you could do `.find.(Title)`

Reply:
nit and loud thinking: Title is an implementation detail, the rendered dom (h1) is not (we have no choice, we render DOM at the end of the day).
If we start testing our own abstractions instead of the actual rendered output, we will end up writing shallow-like bad tests.

Reply:
While I mostly agree with avoiding shallow testing, in this particular case it also means that the drawback by not using `Title` is I can replace the `Title` by a `<h1>` in my component and the test will pass, while it shouldn't since Title also contains specific style.

Reply:
If a use a snapshot for this test:
- We no longer need a selector to test the presence of an element
- The styles appear in the snapshot as we use Styled Components so replacing the `Title` component by a `h1` would break that part.

What do you think?

Reply:
Snapshot tests are not very focused and it's easy to not see all the diffs when we refactor heavily tests. Non-snapshot unit tests allow us to have a more detailed list of use cases that we want to test. For these reasons, we should try to minimize the use of snapshot tests; a good balance is too have a lot of non-snapshot unit tests  + a few snapshot tests to test the default high level states (like smoke integration tests).

About the original issue about selecting using Title or h1, I would still prefer h1 since this is the real thing I want to test the existence of (not some intermediate abstractions).

All in all, both are fine too me.

Reply:
I'll keep the `h1` for now as the other test is also using it, but I'll be happy to investigate a better solution for the tests.